### PR TITLE
STOR-1315: Add e2e workflow for Secrets Store CSI Driver

### PR DESCRIPTION
**Pull request by dobsonj**
[STOR-1315](https://issues.redhat.com//browse/STOR-1315): Add e2e workflow for Secrets Store CSI Driver

depends on https://github.com/openshift/secrets-store-csi-driver-operator/pull/7

cc @openshift/storage

### CODE REVIEW

**Comment by dobsonj** [comment]
/pj-rehearse pull-ci-openshift-secrets-store-csi-driver-operator-main-operator-e2e

**Comment by jsafrane** [comment]
> Suite run returned error: suite "openshift/conformance/serial" does not contain any tests

??

**Comment by dobsonj** [comment]
> > Suite run returned error: suite "openshift/conformance/serial" does not contain any tests
> 
> ??

@jsafrane it's because I restricted the test run [just to SecretsStore tests](https://github.com/openshift/release/blob/21be6a693b8a8bf360efe5ac5184aeafb5f74e02/ci-operator/step-registry/openshift/e2e/azure/csi/secrets-store/openshift-e2e-azure-csi-secrets-store-workflow.yaml#L18) and https://github.com/openshift/origin/pull/28088 hasn't merged yet so there aren't any tests for it to run yet.

I can at least see the e2eprovider image is making it into the pipeline now though
```
              {
                "name": "SECRETS_STORE_E2E_PROVIDER_IMAGE",
                "value": "registry.build04.ci.openshift.org/ci-op-zmmlnllf/pipeline@sha256:e5c694d3b0fdd6705077fcb7af4e8053d483b75d52722a29dd81a49ee6a404e7"
              },
```

I have a few small changes left to make to both PR's but it should work.

**Comment by dobsonj** [comment]
The workflow is doing what I expect now, it's just failing because https://github.com/openshift/origin/pull/28088 needs to merge first.
/hold for https://github.com/openshift/origin/pull/28088
/retitle STOR-1315: Add e2e workflow for Secrets Store CSI Driver

**Comment by dobsonj** [comment]
I changed the workflow to use `make test-e2e` from the operator repo after https://github.com/openshift/secrets-store-csi-driver-operator/pull/7 merged:
https://github.com/openshift/release/blob/d03ca0afa1770e7e87b494800ea5f626e4720f11/ci-operator/step-registry/openshift/e2e/azure/csi/secrets-store/openshift-e2e-azure-csi-secrets-store-workflow.yaml#L10-L18

/unhold
because it's no longer blocked

/retest
because lots of init container failures...??

/pj-rehearse
to see if it works

**Comment by dobsonj** [comment]
/pj-rehearse ack
logs look good:
- [storage-create-csi-secrets-store](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/41655/rehearse-41655-pull-ci-openshift-secrets-store-csi-driver-operator-main-operator-e2e/1686877793108889600/artifacts/operator-e2e/storage-create-csi-secrets-store/build-log.txt)
- [test-e2e](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/41655/rehearse-41655-pull-ci-openshift-secrets-store-csi-driver-operator-main-operator-e2e/1686877793108889600/artifacts/operator-e2e/test-e2e/build-log.txt)
- [storage-destroy-csi-secrets-store](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/41655/rehearse-41655-pull-ci-openshift-secrets-store-csi-driver-operator-main-operator-e2e/1686877793108889600/artifacts/operator-e2e/storage-destroy-csi-secrets-store/build-log.txt)

File: ci-operator/step-registry/storage/create/csi-secrets-store/storage-create-csi-secrets-store-commands.sh [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	fi
+	loop_counter=${loop_counter}+1
+	sleep 5
+done
+
+echo "Waiting for E2E Provider pods to be Ready"
+if ! oc wait pod ${E2E_PROVIDER_GET_ARGS} --for=condition=Ready --timeout=300s; then
+	echo "Wait failed, E2E Provider pods did not reach Ready state"
+	exit 1
+fi
```
Comment (jsafrane) [inline ci-operator/step-registry/storage/create/csi-secrets-store/storage-create-csi-secrets-store-commands.sh:120]: you can wait for DaemonSet numberReady to be equal to nr. of nodes:

`oc -n openshift-cluster-csi-drivers wait --for=jsonpath='{.status.numberReady}'=$NODES daemonset/aws-ebs-csi-driver-node`

BTW, for any error case, you should dump the DaemonSet yaml for debugging.

Comment (dobsonj) [reply ci-operator/step-registry/storage/create/csi-secrets-store/storage-create-csi-secrets-store-commands.sh]: Cool trick, I pushed a new commit with this change.

**Pull request merged**